packages/venia-ui/lib/components/OrderHistoryPage/orderRow.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { mergeClasses } from '../../classify';
+
+import defaultClasses from './orderRow.css';
+import { Price } from '@magento/peregrine';
+import OrderProgressBar from './orderProgressBar';
+import { useOrderRow } from '@magento/peregrine/lib/talons/OrderHistoryPage/useOrderRow';
+import { ChevronDown, ChevronUp } from 'react-feather';
+import Icon from '../Icon';
+import CollapsedImageGallery from './collapsedImageGallery';
+
+const OrderRow = props => {
+    const { order } = props;
+    const {
+        invoices,
+        items,
+        number: orderNumber,
+        order_date: orderDate,
+        shipments,
+        status,
+        total
+    } = order;
+    const { grand_total: grandTotal } = total;
+    const { currency, value: orderTotal } = grandTotal;
+
+    const formattedDate = new Date(orderDate).toLocaleDateString(
+        navigator.locale,
+        {
+            year: 'numeric',
+            month: 'short',
+            day: 'numeric'
+        }
+    );
+
+    const hasInvoice = !!invoices.length;
+    const hasShipment = !!shipments.length;
+    const derivedStatus =
```

Comment:
Wow, that looks convoluted at first.

Reply:
Agreed. Would have loved if `status` was more helpful, but it stays in `Processing` pretty much the entire time; had to use these heuristics to get what UX wanted.